lib/widgets/message_list.dart

```diff
@@ -734,13 +734,17 @@ Future<void> markNarrowAsRead(BuildContext context, Narrow narrow) async {
   int responseCount = 0;
   int updatedCount = 0;
 
-  final apiNarrow = switch (narrow) {
-    // Since there's a database index on is:unread, it's a fast
-    // search query and thus worth using as an optimization
-    // when processing all messages.
-    AllMessagesNarrow() => [ApiNarrowIsUnread()],
-    _                   => narrow.apiEncode(),
-  };
+  // This process can take a long time in narrows with a lot
+  // of history given the server ends up processing a large
+  // number of already read messages. Unconditionally add
+  // [ApiNarrowIsUnread] here to take advantage of a database
+  // index on `is:unread`.
+  // This optimization is also used server side for the
+  // (deprecated) specialized endpoints for marking messages
+  // as read. See `do_mark_stream_messages_as_read` in
+  // `zulip:zerver/actions/message_flags.py`.
```

Comment:
```suggestion
  // Include `is:unread` in the narrow.  That has a database index, so
  // this can be an important optimization in narrows with a lot of history.
  // The server applies the same optimization within the (deprecated)
  // specialized endpoints for marking messages as read; see
  // `do_mark_stream_messages_as_read` in `zulip:zerver/actions/message_flags.py`.
```

I think the main thing I wanted to adjust in this comment is that the other version, with "This process can take a long time" and "the server ends up processing", sounds like it's describing what *does* happen, whereas in fact it's describing what *would* happen if the code worked in a different way.

A direct fix for that could look like "If we used `narrow.apiEncode()` directly, this process could …", or like "This process could … given the server would end up processing … messages. Avoid that by adding …".

(A related, smaller, point is that although "unconditionally" is a highly relevant term for the commit message, it's a bit out of place in the code comment. The reader of the code comment is trying to understand the code as it is, and not starting from how the code previously was, so they're left to wonder what the hypothetical other way of doing this was that would have been conditional.)